src/main/java/com/zaxxer/hikari/HikariDataSource.java

```diff
@@ -159,13 +158,12 @@ public int getLoginTimeout() throws SQLException
       }
 
       if (pool != null) {
-         if (iface.isInstance(pool.getDataSource())) {
-            return (T) pool.getDataSource();
+         final DataSource dataSource = pool.getDataSource();
+         if (iface.isInstance(dataSource)) {
+            return (T) dataSource;
          }
 
-         if (pool.getDataSource() instanceof Wrapper) {
-            return (T) pool.getDataSource().unwrap(iface);
-         }
+         return dataSource.unwrap(iface);
```

Comment:
This seems like a good idea, but the reality is there are drivers out there compiled before Java 6, and calling like this without the *instanceof* check will result in an java.lang.AbstractMethodError -- rather than the more useful exception we throw below.